src/libs/ajax/leonardo/models/runtime-config-models.ts

```diff
@@ -69,3 +69,6 @@ export const isGceConfig = (config: RuntimeConfig): config is GceConfig => {
   const castConfig = config as GceConfig
   return config.cloudService === 'GCE' && castConfig.diskSize !== undefined
 }
+export const isAzureConfig = (config: RuntimeConfig): config is AzureConfig => {
```

Comment:
This is a little hard for my smooth brain to parse - what does `(x: SomeType): x is SomeSubType` do in this case? Is this a way to allow any type of RuntimeConfig to be passed as a param, but only execute the function if the config is an AzureConfig? If it isn't, is the result an error, a boolean `false`, or something else?

Reply:
it's a type assignment function, one which is marked by typescript as being able to discriminate if an input can be a certain type. It'll return true if it is a valid azureconfiguration, and the ESLint can tune into this and properly lint the if statement